redaxo/include/classes/class.rex_sql.inc.php

```diff
@@ -558,7 +558,8 @@ class rex_sql
      */
     /*public*/ function hasNext()
     {
-        return $this->counter != $this->rows;
+        #return $this->counter != $this->rows;
+        return (($this->counter) < ($this->rows-1));
```

Comment:
Shouldnt it be either `counter < rows` or `counter <= rows-1`? Counter is zero based and rows is 1 based..